hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/SelectByKeyDeoptPhysicalRule.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.opt.physical;
+
+import com.hazelcast.jet.sql.impl.opt.OptUtils;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelRule;
+import org.apache.calcite.plan.volcano.RelSubset;
+import org.apache.calcite.rel.RelNode;
+import org.immutables.value.Value;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.jet.sql.impl.opt.Conventions.PHYSICAL;
+
+@Value.Enclosing
+final class SelectByKeyDeoptPhysicalRule extends RelRule<RelRule.Config> {
```

Comment:
This is weird. Ideally we shouldn't have produced a non-working plan in the first place. We probably can't have a rule that will match an element on the root, but instead after logical opt we can look at the whole plan, and convert to select-by-key, if the whole tree is just a single full-scan.